packages/block/src/header.ts

```diff
@@ -254,28 +254,36 @@ export class BlockHeader {
   }
 
   /**
-   * Returns the canonical difficulty for this block.
-   *
-   * @param parentBlockHeader - the header from the parent `Block` of this header
+   * @param parentBlock - The `parentBlock` where we wish to build upon
+   * @param timestamp - The `BN` timestamp which we should calculate the difficulty for
+   * @param number - The `BN` block number. Defaults to the parent blocks' `number` + 1
+   * @param common - The `Common` to use: defaults to the parent blocks` `Common`
    */
-  canonicalDifficulty(parentBlockHeader: BlockHeader): BN {
-    const hardfork = this._getHardfork()
-    const blockTs = this.timestamp
-    const { timestamp: parentTs, difficulty: parentDif } = parentBlockHeader
+
+  public static getCanonicalDifficulty(
+    parentHeader: BlockHeader,
+    timestamp: BN,
+    number?: BN,
+    common?: Common
+  ): BN {
+    const blockTs = timestamp.clone()
+    const { timestamp: parentTs, difficulty: parentDif } = parentHeader
+    const usedCommon = common || parentHeader._common
```

Comment:
You can simply do `common = common || parentHeader._common` here, `common` will already initialized and set to `undefined` if not passed in as a parameter, this would be easier to read in the following and better match with patterns we have in other library parts like the VM.

Reply:
Fixed, fd196353. For some reason I thought that editing call arguments were not possible - but it is! 😄